storey/steps/groupbykey.py

```diff
@@ -0,0 +1,32 @@
+from typing import Optional, Union
+
+from storey.dtypes import Event
+from storey.flow import _Batching
+
+
+class GroupByKey(_Batching):
+    """
+    Groups events by given key (or key extractor function) into lists of up to max_events events. Each emitted list
+    contained max_events events, unless timeout_secs seconds have passed since the first event in the group was
+    received, at which the group is emitted with potentially fewer than max_events
+    event.
+    :param key: The key on which events are grouped by.
+    :type key: str or callable
+    :param max_events: Maximum number of events per emitted batch. Set to None to emit all events in one batch on flow termination.
+    :type max_events: int or None
+    :param timeout_secs: Maximum number of seconds to wait before a batch is emitted.
+    :type timeout_secs: int
```

Comment:
No need for :type when using type annotations.

Reply:
The new to me, I'll remove it